lib/data/multipleBackendGateway.js

```diff
@@ -215,9 +215,9 @@ const multipleBackendGateway = {
     },
 
     objectTagging: (method, key, bucket, objectMD, log, cb) => {
-        const location = objectMD.dataStoreName;
+        const location = objectMD.dataStoreName || undefined;
```

Comment:
Could this instead be `objectMD.controllingLocationConstraint` instead of adding another field `dataStoreName`?

In any case, seems `|| undefined` isn't necessary.